Doc/library/sys.rst

```diff
@@ -1446,9 +1446,13 @@ always available.
      for the Windows console, this only applies when
      :envvar:`PYTHONLEGACYWINDOWSSTDIO` is also set.
 
-   * When interactive, ``stdout`` and ``stderr`` streams are line-buffered.
-     Otherwise, they are block-buffered like regular text files.  You can
-     override this value with the :option:`-u` command-line option.
+   * When interactive, the ``stdout`` stream is line-buffered. Otherwise,
+     it is block-buffered like regular text files.  The ``stderr`` stream
+     is line-buffered in both cases.  You can override this behaviour with
+     the :option:`-u` command-line option.
+
+   .. versionchanged:: 3.9
+      Use line-buffering for non-interactive ``stderr``.
```

Comment:
Would you mind to elaborate what was the behavior previously?

What is the kind of buffering when line-buffering is not used? "Fully buffered" or "unbuffered"?

Reply:
If line-buffering is disabled, sys.stderr is fully buffered.